### PR TITLE
Terraform module for SWOOP Workflow Configuration

### PR DESCRIPTION
**Pull request by geomatician**

### CODE REVIEW

File: modules/workflow_config/workflow_config/workflow_config.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+resource "helm_release" "workflow_config" {
+  name       = "workflow-config"
+  namespace  = "swoop"
```
Comment (hectormachin) [inline modules/workflow_config/workflow_config/workflow_config.tf:3]: pass namespace as an input parameter

File: profiles/core/inputs.tf [hunk 2]
```diff
@@ -1,237 +1,243 @@
-variable deploy_linkerd { 
+variable "deploy_linkerd" {
```
Comment (hectormachin) [inline profiles/core/inputs.tf:1]: not sure why your local formatting is adding these extra quotes around variable names, but can we remove them?

File: profiles/core/inputs.tf [hunk 3]
```diff
@@ -578,25 +584,25 @@ variable "stac_fastapi_secret" {
...
-  default = "0.1.0"
+  default     = "0.1.0"
 }
 
-variable stac_fastapi_additional_configuration_values {
+variable "stac_fastapi_additional_configuration_values" {
   type        = list(string)
   default     = []
   description = "List of values in raw yaml to pass to helm. Values will be merged, in order, as Helm does with multiple -f options."
 }
```
Comment (hectormachin) [inline profiles/core/inputs.tf:603]: were missing the keys and region inputs

```
variable "aws_access_key" {
  description = "AWS Access Key ID"
  type        = string
  default     = ""
  sensitive   = true
}

variable "aws_secret_access_key" {
  description = "AWS Secret Access Key"
  type        = string
  default     = ""
  sensitive   = true
}

variable "aws_region" {
  description = "AWS Region"
  type        = string
  default     = ""
  sensitive   = true
}

```
along with the optional session token

File: modules/workflow_config/workflow_config/inputs.tf [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  default     = ""
+  sensitive   = true
+}
+
+variable "aws_region" {
+  description = "AWS Region"
+  type        = string
+  default     = ""
+  sensitive   = true
+}
```
Comment (hectormachin) [inline modules/workflow_config/workflow_config/inputs.tf:32]: I think we're missing the optional session token that was recently added to the helm chart

File: modules/workflow_config/inputs.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  default     = ""
+  sensitive   = true
+}
+
+variable "aws_region" {
+  description = "AWS Region"
+  type        = string
+  default     = ""
+  sensitive   = true
+}
```
Comment (hectormachin) [inline modules/workflow_config/inputs.tf:38]: missing the session token

File: modules/workflow_config/main.tf [hunk 6]
```diff
@@ -0,0 +1,8 @@
+module "workflow_config" {
+  count     = var.deploy_workflow_config == true ? 1 : 0
+  source    = "./workflow_config"
+  namespace = var.namespace
+
```
Comment (hectormachin) [inline modules/workflow_config/main.tf:5]: missing the module inputs for the keys

File: modules/workflow_config/main.tf [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  namespace_annotations = var.namespace_annotations
+
+}
```
Comment (hectormachin) [inline modules/workflow_config/main.tf:8]: We should create a secret with the access key, secret access key, region, and session token, and pass that secret name to the helm chart

File: profiles/core/inputs.tf (hunk 3 above)
Comment (hectormachin) [reply profiles/core/inputs.tf:603]: we also need to add these at the upper inputs.tf at the top level of the repo in: https://github.com/Element84/filmdrop-k8s-tf-modules/blob/config/inputs.tf

**Review by hectormachin — CHANGES_REQUESTED** [review]

File: modules/workflow_config/main.tf (hunk 7 above)
Comment (geomatician) [reply modules/workflow_config/main.tf:11]: The helm chart itself creates a secret with those keys...  so, why do we need to re-create it in Terraform?

**Review by hectormachin — APPROVED** [review]

**Pull request merged**